storage/src/cache/cachedfile.rs

```diff
@@ -250,19 +250,21 @@ impl BlobCache for FileCacheEntry {
         let mut total_size = 0;
         let mut start = 0;
         while start < pending.len() {
-            let mut end = start + 1;
-            while end < pending.len() && pending[end].id() == pending[end - 1].id() + 1 {
+            let mut end = start;
+
+            while end < pending.len() - 1 && pending[end + 1].id() == pending[end].id() + 1 {
                 end += 1;
```

Comment:
`end` is exclusive in the original code. The patch changes it to be inclusive. How does it fix a panic?

Reply:
> Given that there are no tests to validate the change and it is a critical code path, I need to better understand it before giving it a pass. Can you attach the panic trace that you are seeing? Better some explanation on why it panics and how the patch fixes it.

Please check out the panic backtrace. I will also append one into the commit log.

```
thread 'cache-flusher' panicked at 'index out of bounds: the len is 2 but the index is 2': storage/src/cache/cachedfile.rs:328
   0: <backtrace::capture::Backtrace as core::default::Default>::default
             at /home/gechangwei/.cargo/registry/src/mirrors.sjtug.sjtu.edu.cn-4f7dbcce21e258a2/backtrace-0.3.66/src/capture.rs:410:9
   1: log_panics::Config::install_panic_hook::{{closure}}
             at /home/gechangwei/.cargo/registry/src/mirrors.sjtug.sjtu.edu.cn-4f7dbcce21e258a2/log-panics-2.1.0/src/lib.rs:115:29
   2: std::panicking::rust_panic_with_hook
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:702:17
   3: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:588:13
   4: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/sys_common/backtrace.rs:138:18
   5: rust_begin_unwind
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:584:5
   6: core::panicking::panic_fmt
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/panicking.rs:143:14
   7: core::panicking::panic_bounds_check
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/panicking.rs:84:5
   8: <usize as core::slice::index::SliceIndex<[T]>>::index
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/slice/index.rs:241:10
   9: core::slice::index::<impl core::ops::index::Index<I> for [T]>::index
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/slice/index.rs:18:9
  10: <alloc::vec::Vec<T,A> as core::ops::index::Index<I>>::index
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/alloc/src/vec/mod.rs:2533:9
  11: <nydus_storage::cache::cachedfile::FileCacheEntry as nydus_storage::cache::BlobCache>::prefetch_range
             at /data00/home/gechangwei/git_repo/nydus-rs/storage/src/cache/cachedfile.rs:328:28
  12: nydus_storage::cache::worker::AsyncWorkerMgr::handle_fs_prefetch_request
             at /data00/home/gechangwei/git_repo/nydus-rs/storage/src/cache/worker.rs:392:9
  13: nydus_storage::cache::worker::AsyncWorkerMgr::handle_prefetch_requests::{{closure}}::{{closure}}
             at /data00/home/gechangwei/git_repo/nydus-rs/storage/src/cache/worker.rs:268:37
  14: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /home/gechangwei/.cargo/registry/src/mirrors.sjtug.sjtu.edu.cn-4f7dbcce21e258a2/tokio-1.20.0/src/runtime/blocking/task.rs:42:21
  15: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /home/gechangwei/.cargo/registry/src/mirrors.sjtug.sjtu.edu.cn-4f7dbcce21e258a2/tokio-1.20.0/src/runtime/task/core.rs:165:17
```

Reply:
Please DO NOT edit my comment :) Your comments can be raised as usual.